base/src/main/java/org/aya/resolve/context/ModuleContext.java

```diff
@@ -1,155 +1,226 @@
-    var result = definitions().getOrNull(name);
-    if (result == null) return null;
-    if (result.size() == 1) return result.iterator().next().getValue();
-    return reportAndThrow(new NameProblem.AmbiguousNameError(
-      name,
-      result.keysView().toImmutableSeq(),
-      sourcePos));
-  }
+public sealed interface ModuleContext extends ModuleLikeContext permits NoExportContext, PhysicalModuleContext {
+  /**
+   * All available symbols in this context<br>
+   * {@code Unqualified -> (Module Name -> TopLevel)}<br>
+   * It says an {@link ContextUnit.TopLevel} can be referred by {@code {Module Name}::{Unqualified}}
+   */
+  @Override @NotNull MutableModuleSymbol<ContextUnit.TopLevel> symbols();
+
+  /**
+   * All imported modules in this context.<br/>
+   * {@code Qualified Module -> Module Export}
+   *
+   * @apiNote empty list => this module
+   * @implNote This module should be automatically imported.
+   */
+  @Override @NotNull MutableMap<ModulePath, ModuleExport> modules();
+
+  @NotNull MutableModuleExport thisModule();
 
-  @Override
-  default @Nullable AnyVar getQualifiedLocalMaybe(
-    @NotNull ImmutableSeq<@NotNull String> modName,
-    @NotNull String name,
-    @NotNull SourcePos sourcePos) {
-    var mod = modules().getOrNull(modName);
-    if (mod == null) return null;
-    var ref = mod.getOrNull(name);
-    if (ref == null) reportAndThrow(new NameProblem.QualifiedNameNotFoundError(modName, name, sourcePos));
-    return ref;
-  }
-
-  @Override default @Nullable ModuleExport getModuleLocalMaybe(@NotNull ImmutableSeq<String> modName) {
-    return modules().getOrNull(modName);
+  default void importModule(
+    @NotNull ModuleLikeContext module,
+    @NotNull Stmt.Accessibility accessibility,
+    @NotNull SourcePos sourcePos
+  ) {
+    importModuleExports(ModulePath.ofQualified(module.moduleName()), module.exports(), accessibility, sourcePos);
   }
 
-  default void importModules(
-    @NotNull ImmutableSeq<String> modName,
+  /**
+   * Import the whole module (including itself and re-exports)
+   *
+   * @see ModuleContext#importModule(ModulePath.Qualified, ModuleExport, Stmt.Accessibility, SourcePos)
+   */
+  default void importModuleExports(
+    @NotNull ModulePath.Qualified modName,
+    @NotNull Map<ModulePath, ModuleExport> module,
     @NotNull Stmt.Accessibility accessibility,
-    @NotNull MutableMap<ImmutableSeq<String>, ModuleExport> module,
     @NotNull SourcePos sourcePos
   ) {
-    module.forEach((name, mod) -> importModule(accessibility, sourcePos, modName.concat(name), mod));
+    module.forEach((name, mod) -> importModule(modName.concat(name), mod, accessibility, sourcePos));
   }
 
+  /**
+   * Importing one module export.
+   *
+   * @param accessibility of importing, re-export if public
+   * @param componentName the name of the module
+   * @param moduleExport  the module
+   */
   default void importModule(
```

Comment:
`importModuleExports` imports everything from a module according to your comment.

`importModule` imports only exported things from a module.

I am confused by their names.